contracts/nns/contract.go

```diff
@@ -637,6 +637,28 @@ func GetAllRecords(name string) iterator.Iterator {
 	return getAllRecords(ctx, name, fragments)
 }
 
+// HasTXTRecord checks if a record of the TXT type exists
+// for the given domain. The name MUST NOT be a TLD.
+func HasTXTRecord(name string, data string) bool {
+	fragments := std.StringSplit(name, ".")
+	if len(fragments) == 1 {
+		panic("token not found")
+	}
+
+	ctx := storage.GetReadOnlyContext()
+	records := getAllRecords(ctx, name, fragments)
```

Comment:
Given `maxRecordID` of 15 (22be33c75243f0cc132cbbdf52a0b291d9b7d143) this limits us to 16 elements which is just not enough for the purpose (consider storing 500 addresses). We need to think about it.

Technically what we need for ideal solution is just some `domain_hash + address` key which can have arbitrary number of addresses and iis trivial to `Get()`, not iterating over the whole set. But this doesn't fit into NNS model.